keras_cv/layers/preprocessing/color_jitter.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+from tensorflow.keras import layers
+
+
+class ColorJitter(layers.Layer):
```

Comment:
@LukeWood , since this is a layer that combines all the color related augmentation. Should we have this or prefer the RandomAugmentLayer (I think this layer is a subset of RandomAugmentLayer )

Reply:
I think this is distinct enough to include.   I do wonder if separating out these operations into 3 separate layers and recommending to users to use them in a chain is a better api.

Reply:
basically, we would have:

```
class ColorJitter():
    def __init__(self):
        self.contrast = RandomContrast()
        self.saturation = RandomSaturation()
       # ... etc  
    def call(self, inputs):
       # can chain as many sub-layers as needed here
        return self.contrast(self.saturation(inputs))
```

This would be more consistent with our design decision for RandAugment.  @innat Lets split this into sub-layers.

Reply:
@LukeWood are you suggesting using the existing layers to build color jitter? random_[contrast](https://www.tensorflow.org/api_docs/python/tf/keras/layers/RandomContrast)/[brightness](https://www.tensorflow.org/api_docs/python/tf/keras/layers/RandomBrightness)/[saturation](https://github.com/keras-team/keras-cv/pull/221)/hue? I think now, only a hue is needed to implement.

Reply:
Yep!  Exactly!  This way we can reuse the logic

Reply:
@LukeWood that makes sense. That would be great. 
- [RandomContrast](https://www.tensorflow.org/api_docs/python/tf/keras/layers/RandomContrast)
- [RandonBrightness](https://www.tensorflow.org/api_docs/python/tf/keras/layers/RandomBrightness)
- [RandomSaturation](https://github.com/keras-team/keras-cv/pull/221)
- [RandaomHue](https://github.com/keras-team/keras-cv/pull/231)

cc. @qlzh727